packages/live-queries/cardstack/service.js

```diff
@@ -0,0 +1,104 @@
+const log = require('@cardstack/logger')('cardstack/live-queries');
+const { declareInjections } = require('@cardstack/di');
+
+const DEFAULT_SOCKET_IO_PORT = 3100;
+const DEFAULT_SOCKET_IO_PATH = '/';
+
+module.exports = declareInjections({
+  plugins: 'hub:plugins',
+  indexers: 'hub:indexers'
+},
+class LiveQueryService {
+  static async create({plugins, indexers}) {
+    let instance = new this();
+
+    // This awkward dance is just to get our own plugin config. We also
+    // have the dummy middleware, just to get this file loaded so it
+    // can start accepting socket connections. We should just add an
+    // initializer feature type that gets run on startup, and passed its
+    // plugin config.
+    // We're happy to start the server in the background though, hence .then()
+    plugins.active().then(async function(configured) {
+      let ourConfig = configured.describe('@cardstack/live-queries');
+      let socketPath = ourConfig.attributes['socket-path'] || DEFAULT_SOCKET_IO_PATH;
+      let socketPort = ourConfig.attributes['socket-port'] || DEFAULT_SOCKET_IO_PORT;
+
+      instance.start({socketPath, socketPort});
+    });
+
+    instance.listenToIndexer(indexers.events);
+
+    return instance;
+  }
+
+  constructor() {
+    this._client_subscriptions = new WeakMap();
+  }
+
+  async start({socketPath, socketPort}) {
+    log.info(`starting socket.io on port ${socketPort} at path ${socketPath}`);
+    this.server = require('socket.io')(socketPort, {
+      path: socketPath,
+      serveClient: false
+    });
+
+
+    this.namespace = this.server.of('/live-queries');
+    this.namespace.on("connection", (socket) => {
+      log.debug(`Received a connection, socket id '${socket.id}'`);
+
+      let subs = new ClientSubscriptionSet(socket);
+      this._client_subscriptions.set(socket, subs);
+
+      socket.on('query-subscribe', subs.subscribe.bind(subs));
+      socket.on('query-unsubscribe', subs.subscribe.bind(subs));
+    });
+  }
+
+  listenToIndexer(indexEvents) {
+    indexEvents.on('index_update', () => this._invalidateAll());
```

Comment:
maybe not for this PR, but as a future direction, I wonder if we should introduce a new hub feature that plugins can use that will allow a plugin to determine which models to invalidate based on a hint being passed within the index event. maybe it's called "live-indexer" feature?

Reply:
yeah, Aaron and I discussed this. The plan is to make sure that the client-to-server API provides the full query, giving us a place in the server where we could steadily make the invalidation smarter by understanding more queries, while erring on the side of over-invalidating.